### ISSUE TITLE
hyper par settings: partial matching hints

### ISSUE THREAD

**Issue by berndbischl** [mlr-org/mlr#1050]
if a user enters a wrongly typed param name, display, instead of only the error, a hint which other name matches most closely.

like:
"did you mean ntrees instead of n.tree?"

**Comment by ja-thomas** [comment]
I really like that!

**Comment by berndbischl** [comment]
me too :)
dont know why we didnt see this before. this was a suggesting from a participant of my ml course
(thanks, wolfgang!)

**Comment by wolfganglederer** [comment]
You are welcome.

**Comment by wolfganglederer** [comment]
You are welcome.

**Comment by wolfganglederer** [comment]
You are welcome.

**Comment by berndbischl** [comment]
> You are welcome.

:)

**Comment by ja-thomas** [comment]
we could use the [levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance), which is already implemented in `adist` in Base-r.

e.g.

```
pars = names(getParamSet("classif.ranger")$pars)

pars[which.min(adist(pars), "n.tree"))]
pars[which.min(adist(pars), "trees"))]
pars[which.min(adist(pars), "nrounds"))]
pars[which.min(adist(pars), "iters"))]
pars[which.min(adist(pars), "numberoftree"))]
```

**Comment by MasonGallo** [comment]
+1 

`adist` in {utils} does fuzzy matching - I can do a PR soon if no one else does

**Comment by berndbischl** [comment]
@MasonGallo 
do a PR pls.

**Comment by berndbischl** [comment]
@MasonGallo 
solved this (nearly) in PR #1060
